### PR TITLE
Upgrade addon to Ember 3.4

### PR DESCRIPTION
**Pull request by patcoll**
[NEEDS REBASE AFTER OTHER PRs ARE MERGED.]

Done as an incremental step toward a higher Ember version.

- Remove `withVariation` test helper
- Remove Ember 2.x from ember-try config to only target higher versions
- Fix Ember imports everywhere
- Fix code that variation babel transform generates
- Fix tests

## How to test

- Run `yarn run test:ci` and all tests should pass.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,10 +4,61 @@ module.exports = {
...
+      files: [
+        'nodetests/**/*.js'
+      ],
+      parserOptions: {
+        sourceType: 'module'
+      },
+      env: {
+        browser: false,
+        node: true
+      }
```
Comment (patcoll) [inline .eslintrc.js]: Added for tests in this folder

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# compiled output
+/dist/
+/tmp/
+
+# dependencies
+/bower_components/
+
+# misc
+/coverage/
+/nodetests/__fixtures__/
```
Comment (patcoll) [inline .eslintignore]: Ignore fixture files since they have invalid syntax

File: .travis.yml [hunk 3]
```diff
@@ -1,44 +1,54 @@
...
+      script:
+        - npm run lint:hbs
+        - npm run lint:js
+        - npm test
+
+    # we recommend new addons test the current and previous LTS
+    # as well as latest stable release (bonus points to beta/canary)
+    - stage: "Additional Tests"
+      env: EMBER_TRY_SCENARIO=ember-lts-2.16
+    - env: EMBER_TRY_SCENARIO=ember-lts-2.18
```
Comment (patcoll) [inline .travis.yml]: TODO: Remove Ember 2.x scenarios

File: tests/dummy/app/controllers/login.js [hunk 4]
```diff
@@ -38,14 +36,4 @@ export default Controller.extend({
...
-
-  // BEGIN-SNIPPET as-js-variation-helper
-  foo: computed(function () {
-    if (variation('apply-discount')) {
-      return '99';
-    }
-
-    return '199';
-  })
-  // END-SNIPPET
```
Comment (patcoll) [inline tests/dummy/app/controllers/login.js]: Removed `as-js-variation-helper`

File: tests/dummy/app/templates/login.hbs [hunk 5]
```diff
@@ -41,15 +40,8 @@ PRICE: £ {{anotherPrice}}
...
   PRICE: £ 199
 {{/with}}
 {{!--END-SNIPPET--}}
-
-<br>
-<br>
-
-{{code-snippet name="as-js-variation-helper.js"}}
-
-PRICE: £ {{foo}}
```
Comment (patcoll) [inline tests/dummy/app/templates/login.hbs]: Removed `as-js-variation-helper`

**Comment by achambers** [comment]
Gaaahhhhh, I missed this. So, last week I was working on upgrading to 3.11 (pulling out those other separate small PR's along the way).

So, apologies for the effort you went to mate. Got our wires crossed I think. May as well go with 3.11 as it's the latest version.

I'll look through this PR to ensure there's nothing there that I should have in mine. I imagine they are largely the same however.

**Pull request closed without merging**